src/zenml/materializers/base_materializer.py

```diff
@@ -36,10 +37,13 @@ def __new__(
             Type["BaseMaterializer"], super().__new__(mcs, name, bases, dct)
         )
         if name != "BaseMaterializer":
-            assert cls.ASSOCIATED_TYPES, (
-                "You should specify a list of ASSOCIATED_TYPES when creating a "
-                "Materializer!"
-            )
+            if not cls.ASSOCIATED_TYPES:
```

Comment:
Should we perhaps add a type check for the `ASSOCIATED_TYPES` and `ASSOCIATED_ARTIFACT_TYPES`.

Right now, I can do this:
```python
from zenml.materializers.base_materializer import BaseMaterializer


class MyMaterializer(BaseMaterializer):
    ASSOCIATED_TYPES = "random thing"
    ASSOCIATED_ARTIFACT_TYPES = 3
```
and it does not fail and PyCharm is giving me no warnings.

Reply:
Very good idea, I'll add that! 
I will limit `ASSOCIATED_ARTIFACT_TYPES` to be subclasses of `BaseArtifacts`. Do you have a suggestion how we can limit the `ASSOCIATED_TYPES` other than checking if it's a class?

Reply:
Checking if it's a class should be fine for now. Perhaps we can also check whether both these variables are defined as tuples.

Reply:
Alright added messages for those two cases

Reply:
What do you think about checking whether they are tuples? (or some form of iterables). Otherwise, if I am not missing anything, line 47 in the `base_materializer.py` would fail, right?

Reply:
I'm a little worried that we're going down the wrong path here. I don't think it's reasonable to check the type of every attribute and parameter, so I feel like we should limit it to places where it is really helpful and people might go wrong.
I can see someone putting a non `BaseArtifact` subclass as an associated artifact type, but if they inherit from `BaseMaterializer` and just overwrite the attribute with a completely unrelated type that's on them and I wouldn't care about that. (The error message will probably say something like `__ is not iterable` which is descriptive enough).
Otherwise we'll start implementing checks that `Integration.REQUIREMENTS` is actually a list, and every element is a string, .... 90% of our code would just be emulating a statically typed language with python